application/src/main/java/org/togetherjava/tjbot/commands/moderation/BanCommand.java

```diff
@@ -0,0 +1,158 @@
+            .flatMap(sendDmResult -> {
+                logger.info(
+                        "'{}' ({}) banned the user '{}' ({}) from guild '{}' and deleted their message history of the last {} days, for reason '{}'.",
+                        author.getUser().getAsTag(), author.getId(), target.getAsTag(),
+                        target.getId(), guild.getName(), deleteHistoryDays, reason);
+
+                return guild.ban(target, deleteHistoryDays, reason)
+                    .map(banResult -> sendDmResult.isSuccess());
+            })
+            .map(hasSentDm -> {
+                String dmNotice =
+                        Boolean.TRUE.equals(hasSentDm) ? "" : "(Unable to send them a DM.)";
+                return ModerationUtils.createActionResponse(author.getUser(),
+                        ModerationUtils.Action.BAN, target, dmNotice, reason);
+            })
+            .flatMap(event::replyEmbeds);
+    }
+
+    private static Optional<RestAction<InteractionHook>> handleNotAlreadyBannedResponse(
+            @NotNull Throwable alreadyBannedFailure, @NotNull Interaction event,
+            @NotNull Guild guild, @NotNull User target) {
+        if (alreadyBannedFailure instanceof ErrorResponseException errorResponseException) {
+            if (errorResponseException.getErrorResponse() == ErrorResponse.UNKNOWN_BAN) {
+                return Optional.empty();
+            }
+
+            if (errorResponseException.getErrorResponse() == ErrorResponse.MISSING_PERMISSIONS) {
+                logger.error("The bot does not have the '{}' permission on the guild '{}'.",
+                        Permission.BAN_MEMBERS, guild.getName());
+                return Optional.of(event.reply(
+                        "I can not ban users in this guild since I do not have the %s permission."
+                            .formatted(Permission.BAN_MEMBERS))
+                    .setEphemeral(true));
+            }
+        }
+        logger.warn("Something unexpected went wrong while trying to ban the user '{}'.",
+                target.getAsTag(), alreadyBannedFailure);
+        return Optional.of(event.reply("Failed to ban the user due to an unexpected problem.")
+            .setEphemeral(true));
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandEvent event) {
+        OptionMapping targetOption =
+                Objects.requireNonNull(event.getOption(TARGET_OPTION), "The target is null");
+        Member targetMember = targetOption.getAsMember();
+        User targetUser = targetOption.getAsUser();
+        Member author = Objects.requireNonNull(event.getMember(), "The author is null");
+        String reason = Objects.requireNonNull(event.getOption(REASON_OPTION), "The reason is null")
+            .getAsString();
+
+        Guild guild = Objects.requireNonNull(event.getGuild());
+        Member bot = guild.getSelfMember();
+
+        // Member doesn't exist if attempting to ban a user who is not part of the guild.
```

Comment:
Might suggest removing this. I added but forget to remove it
```suggestion
```

Reply:
I like it, why do you want to remove it? It explains why the `targetMember != null` check is necessary and prevents that readers trip over it.

Reply:
oh ok